src/utils/price_estimator.ts

```diff
@@ -0,0 +1,4 @@
+export interface PriceEstimator {
```

Comment:
I think AR should be in the name, i.e. `ARDataPriceEstimator`. Then the implementation could be named something like `ARDataPriceRegressionEstimator` since it uses a regression to do estimations.